cargo-espflash/src/main.rs

```diff
@@ -346,20 +369,32 @@ fn flash(args: FlashArgs, config: &Config) -> Result<()> {
             args.flash_args.image,
             &flash_config,
             config,
-            build_ctx.bootloader_path.as_deref(),
-            build_ctx.partition_table_path.as_deref(),
+            chip,
+            target_xtal_freq,
+        );
+        let image_format = make_image_format(
+            &elf_data,
+            &flash_data,
+            args.format,
+            config,
+            Some(args.esp_idf_format_args),
+            build_ctx.bootloader_path,
+            build_ctx.partition_table_path,
         )?;
 
-        if args.flash_args.erase_parts.is_some() || args.flash_args.erase_data_parts.is_some() {
-            erase_partitions(
-                &mut flasher,
-                flash_data.partition_table.clone(),
-                args.flash_args.erase_parts,
-                args.flash_args.erase_data_parts,
-            )?;
+        // If using ESP-IDF image format, check if we need to erase partitions.
+        if let ImageFormat::EspIdf(idf_format) = &image_format {
+            if args.flash_args.erase_parts.is_some() || args.flash_args.erase_data_parts.is_some() {
+                erase_partitions(
+                    &mut flasher,
+                    Some(idf_format.partition_table.clone()),
+                    args.flash_args.erase_parts,
+                    args.flash_args.erase_data_parts,
+                )?;
+            }
```

Comment:
This code snippet is (and was) duplicated in both `cargo-espflash` and `espflash`, shall we do a `maybe_erase_parts` (?) method? 

Also, `check_esp_idf_args` could return a `bool` that could indicate if we need to erase any partition to avoid double checking